modules/cyusb-tools/src/cygpioset.cpp

```diff
@@ -0,0 +1,344 @@
+
+  while (opt != -1)
+  {
+    switch (opt)
+    {
+      case 0:  // long option
+      {
+        const char* p_cstr_long_argument_name = a_long_options[option_index].name;
+        DEBUGLOG("[DEBUG] Option " << p_cstr_long_argument_name << " was provided");
+        if (::optarg == nullptr)
+        {
+          DEBUGLOG(std::endl);
+          /*
+          if (std::strcmp(p_cstr_long_argument_name, "help") == 0)
+          {
+            enum_exit_code = ExitCodeEnum::PRINT_USAGE_AND_EXIT;
+          }
+          else
+          */
+          {
+            std::cout << "[ERROR] Argument " << p_cstr_long_argument_name << " unrecognized--Aborting!" << std::endl;
+            enum_exit_code = ExitCodeEnum::CMDLINE_PARSE_ERROR;
+          }
+        }
+        else  // if(::optarg != nullptr)
+        {
+          DEBUGLOG(" with argument " << ::optarg << std::endl);
+          std::cout << "[ERROR] Error parsing command-line argument \"--" << p_cstr_long_argument_name
+                    << " with argument " << ::optarg << "\"--Aborting!" << std::endl;
+          enum_exit_code = ExitCodeEnum::CMDLINE_PARSE_ERROR;
+        }
+      }
+      break;
+
+      case 1:  // regular non-option argument
+      {
+        const std::string str_optarg{ ::optarg };
+        std::istringstream istream_optarg{ str_optarg };
+
+        if (regular_arg_index == 0)  // gpiochip-num
+        {
+          if (str_optarg.compare(0, std::strlen("gpiochip"), "gpiochip") == 0)
+          {
+            enum_exit_code = ::parse_arg_as_integer("gpiochip-num", ::optarg + std::strlen("gpiochip"), r_app_settings_out.gpiochip_num);
+          }
+          else
+          {
+            std::cout << "[ERROR] Argument " << ::optarg << " unrecognized--Aborting!" << std::endl;
+            enum_exit_code = ExitCodeEnum::UNKNOWN_CMDLINE_ARG;
+          }
+        }
+        else  // offsetX=[0|1]
+        {
+          ::OffsetValuePair offset_value_pair;
+
+          std::string str_offset_out;
+          if (std::getline(istream_optarg, str_offset_out, '='))
+          {
+            enum_exit_code = ::parse_arg_as_integer("offset", str_offset_out.c_str(), offset_value_pair.first);
+          }
+
+          if (enum_exit_code == ExitCodeEnum::SUCCESS)
+          {
+            std::string str_value_out;
+            if (std::getline(istream_optarg, str_value_out))
+            {
+              enum_exit_code = ::parse_arg_as_integer("value", str_value_out.c_str(), offset_value_pair.second);
+            }
+          }
+
+          if (enum_exit_code == ExitCodeEnum::SUCCESS)
+          {
+            r_app_settings_out.offset_value_pair_array.push_back(offset_value_pair);
+          }
+        }
+
+        ++regular_arg_index;
+      }
+      break;
+
+      case 'h':
+        DEBUGLOG("[DEBUG] Option h was provided" << std::endl);
+        enum_exit_code = ExitCodeEnum::PRINT_USAGE_AND_EXIT;
+        break;
+
+        /*
```

Comment:
Are these potential future features?

Reply:
These are demonstrative/testing comments and logging for features that I was going add (like -m, -s, -u) but i ran out of time.  This pattern is for parsing simple string args, integer args, or just bool optoin flags.  Fine with removing, but then if we add later we'll have to figure this pattern out again